### DISCUSSION TITLE
Lua init script for an extension?

### DISCUSSION THREAD

**Discussion by paulwkc** [quarto-dev/quarto-cli#11541] [Q&A]
### Description

Within an extension, I want to include some global Lua functions and update package.path for all lua filter files.

In pandoc, I can use init.lua (https://pandoc.org/lua-filters.html#lua-interpreter-initialization).

How to do that in quarto ? Can I have an init.lua in an extension ?

**Comment by cscheid** [comment]
`local utils = require("relative/path-to/file.lua")` should work.

**Reply by paulwkc** [reply]
It seems some confusion here, or correct me if I'm wrong.

Explicitly, can we do below in quarto ?

1. in `init.lua` :
   ```lua
   -- 1. provide some global functions, like custom_debug() for filter1.lua
   -- 2. add full-path-to/my-lib/ to package.path, for util-glob.lua below , and like libapi1.lua in filter2.lua
   require 'util-glob'
   ```
2. in `filter1.lua` , use custom_debug directly :
   ```lua
   local function func1()
     custom_debug(...)
   end
   ...
   ```
3. in `filter2.lua` , require package directly :
   ```lua
   require 'libapi1'
   ...
   ```
